overlord/hookstate/hookmgr.go

```diff
@@ -89,9 +90,14 @@ func Manager(s *state.State) (*HookManager, error) {
 }
 
 // HookTask returns a task that will run the specified hook.
-func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string) *state.Task {
+func HookTask(s *state.State, taskSummary, snapName string, revision snap.Revision, hookName string, payload interface{}) *state.Task {
```

Comment:
When the user requests that a given snap have its configuration updated, we can't just change it globally (in the state) since it may be an invalid config, or the hook may fail. We have to be able to hand it off to the hook to validate it. Once that's done we can save it (with any modifications the hook might make).

Since we can't save it globally, there has to be some other way to get the data from the set command to the hook. Since we don't know when the hook will run, as far as I can tell we have two options:

1) Create some sort of global map in the config manager of requested config patches indexed by... something. But that requires the hook handler to know that index necessary to get the config anyway. Not to mention we already have a way to store information like this in the task, we leads me to:

2) Store the config patches in the task itself. Which is what the payload is here: a generic way to stuff a payload into the hook task.

It doesn't have to be part of HookTask (i.e. the set command could just put that data in), but then the HookManager wouldn't know about it and couldn't extract them into the HookSetup, so the handlers would need access to the raw task in the Context (which we specifically hid from them).

Do you see a better way to do this?